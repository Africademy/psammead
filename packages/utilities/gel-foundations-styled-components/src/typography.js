import { css } from 'styled-components';
import { MEDIA_QUERY_TYPOGRAPHY } from '@bbc/gel-constants/breakpoints';

export const GEL_ATLAS = css`
  font-size: 4.875em;
  line-height: 5.25rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    font-size: 6em;
    line-height: 6.5rem;
  }

  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 8.75em;
    line-height: 9.25rem;
  }
`;

export const GEL_ELEPHANT = css`
  font-size: 3.75em;
  line-height: 4rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    font-size: 4.875em;
    line-height: 5.25rem;
  }

  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 7.25em;
    line-height: 7.75rem;
  }
`;

export const GEL_IMPERIAL = css`
  font-size: 3.125em;
  line-height: 3.375rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    font-size: 4em;
    line-height: 4.5rem;
  }

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    font-size: 6em;
    line-height: 6.5rem;
  }
`;

export const GEL_ROYAL = css`
  font-size: 2.5em;
  line-height: 2.75rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    font-size: 3.25em;
    line-height: 3.75rem;
  }

  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 4.75em;
    line-height: 5.25rem;
  }
`;

export const GEL_FOOLSCAP = css`
  font-size: 2em;
  line-height: 2.25rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    font-size: 2.5em;
    line-height: 2.75rem;
  }

  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 3.5em;
    line-height: 3.75rem;
  }
`;

export const GEL_CANON = css`
  font-size: 1.75em;
  line-height: 2rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    font-size: 2em;
    line-height: 2.25rem;
  }

  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 2.75em;
    line-height: 3rem;
  }
`;

export const GEL_TRAFALGAR = css`
  font-size: 1.25em;
  line-height: 1.5rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    font-size: 1.5em;
    line-height: 1.75rem;
  }

  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 2em;
    line-height: 2.25rem;
  }
`;

export const GEL_PARAGON = css`
  font-size: 1.25em;
  line-height: 1.5rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    font-size: 1.375em;
    line-height: 1.625rem;
  }

  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 1.75em;
    line-height: 2rem;
  }
`;

export const GEL_DOUBLE_PICA = css`
  font-size: 1.25em;
  line-height: 1.5rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    line-height: 1.5rem;
  }

  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 1.5em;
    line-height: 1.75rem;
  }
`;

export const GEL_GREAT_PRIMER = css`
  font-size: 1.125em;
  line-height: 1.375rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    line-height: 1.375rem;
  }

  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 1.25em;
    line-height: 1.5rem;
  }
`;

export const GEL_BODY_COPY = css`
  font-size: 0.9375em;
  line-height: 1.25rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_AND_LARGER} {
    font-size: 1em;
    line-height: 1.375rem;
  }
`;

export const GEL_PICA = css`
  font-size: 0.9375em;
  line-height: 1.25rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_AND_LARGER} {
    font-size: 1em;
    line-height: 1.25rem;
  }
`;

export const GEL_LONG_PRIMER = css`
  font-size: 0.9375em;
  line-height: 1.125rem;

  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 0.875em;
  }
`;

export const GEL_BREVIER = css`
  font-size: 0.875em;
  line-height: 1rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    line-height: 1.125rem;
  }

  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 0.8125em;
  }
`;

export const GEL_MINION = css`
  font-size: 0.75em;
  line-height: 1em;
`;
