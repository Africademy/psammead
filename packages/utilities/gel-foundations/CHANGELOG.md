# Gel Foundations Changelog

| Version | Description |
|---------|-------------|
| 0.1.0   | [PR#XX](https://github.com/BBC-News/psammead/pull/XX) Create initial package, pulled in from gel-constants and gel-foundations-styled-components packages. |
