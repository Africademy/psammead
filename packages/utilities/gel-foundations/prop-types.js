module.exports = require('./dist/prop-types'); // eslint-disable-line import/no-unresolved
