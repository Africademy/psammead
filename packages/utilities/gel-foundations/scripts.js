/* eslint-disable global-require */
module.exports = {
  news: require('./dist/scripts/news').default,
  arabic: require('./dist/scripts/arabic').default,
  burmese: require('./dist/scripts/burmese').default,
  cjkThai: require('./dist/scripts/cjk_thai').default,
  hindiNepaliBengali: require('./dist/scripts/hindi_nepali_bengali').default,
  latin: require('./dist/scripts/arabic').default,
  mundo: require('./dist/scripts/arabic').default,
  sinhala: require('./dist/scripts/sinhala').default,
  tamil: require('./dist/scripts/tamil').default,
};
