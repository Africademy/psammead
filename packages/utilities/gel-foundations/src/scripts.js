/* eslint-disable global-require */
module.exports = {
  latin: require('../dist/scripts/latin-and-cyrillic').default,
  cyrillic: require('../dist/scripts/latin-and-cyrillic').default,
  latinDiacritics: require('../dist/scripts/latin-with-diacritics').default,
  arabic: require('../dist/scripts/arabic').default,
  arabicPashto: require('../dist/scripts/arabic-pashto').default,
  bengali: require('../dist/scripts/bengali').default,
  burmese: require('../dist/scripts/burmese').default,
  hindi: require('../dist/scripts/devanagari-and-gurmukhi').default,
  nepali: require('../dist/scripts/devanagari-and-gurmukhi').default,
  sinhalese: require('../dist/scripts/sinhalese').default,
  tamil: require('../dist/scripts/tamil').default,
  thai: require('../dist/scripts/thai').default,
  chinese: require('../dist/scripts/no-ascenders-or-descenders').default,
  korean: require('../dist/scripts/no-ascenders-or-descenders').default,
  ethiopic: require('../dist/scripts/no-ascenders-or-descenders').default,
};
