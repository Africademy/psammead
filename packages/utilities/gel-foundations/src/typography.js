import { MEDIA_QUERY_TYPOGRAPHY } from '../breakpoints';

const fontFamilyBase = ', Helvetica, Arial, sans-serif';

export const GEL_FF_REITH_SERIF = `ReithSerif${fontFamilyBase}`;
export const GEL_FF_REITH_SANS = `ReithSans${fontFamilyBase}`;
export const GEL_FF_REITH_SANS_COND = `ReithSansCondensed${fontFamilyBase}`;

function getTypeSizes(typogStyleName, script = 'news') {
  // eslint-disable-next-line global-require, import/no-dynamic-require
  const importedScript = require(`./scripts/${script}.json`);
  const typographyStyle = importedScript[typogStyleName];

  const GROUP_A_FONT_SIZE_PX = typographyStyle.groupA.fontSize;
  const GROUP_A_FONT_SIZE = `${GROUP_A_FONT_SIZE_PX / 16}rem`;
  const GROUP_A_LINE_HEIGHT_PX = typographyStyle.groupA.lineHeight;
  const GROUP_A_LINE_HEIGHT = `${GROUP_A_LINE_HEIGHT_PX / 16}rem`;

  const GROUP_B_FONT_SIZE_PX = typographyStyle.groupB.fontSize;
  const GROUP_B_FONT_SIZE = `${GROUP_B_FONT_SIZE_PX / 16}rem`;
  const GROUP_B_LINE_HEIGHT_PX = typographyStyle.groupB.lineHeight;
  const GROUP_B_LINE_HEIGHT = `${GROUP_B_LINE_HEIGHT_PX / 16}rem`;

  const GROUP_D_FONT_SIZE_PX = typographyStyle.groupD.fontSize;
  const GROUP_D_FONT_SIZE = `${GROUP_D_FONT_SIZE_PX / 16}rem`;
  const GROUP_D_LINE_HEIGHT_PX = typographyStyle.groupD.lineHeight;
  const GROUP_D_LINE_HEIGHT = `${GROUP_D_LINE_HEIGHT_PX / 16}rem`;

  return `
        font-size: ${GROUP_A_FONT_SIZE};
        line-height: ${GROUP_A_LINE_HEIGHT};
   
        ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
          font-size: ${GROUP_B_FONT_SIZE};
          line-height: ${GROUP_B_LINE_HEIGHT};
        }
   
        ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
          font-size: ${GROUP_D_FONT_SIZE};
          line-height: ${GROUP_D_LINE_HEIGHT};
        }
      `;
}

export const getCanon = script => getTypeSizes('canon', script);

export const getTrafalgar = script => getTypeSizes('trafalgar', script);

export const getParagon = script => getTypeSizes('paragon', script);

export const getDoublePica = script => getTypeSizes('doublePica', script);

export const getGreatPrimer = script => getTypeSizes('greatPrimer', script);

export const getPica = script => getTypeSizes('pica', script);

export const getLongPrimer = script => getTypeSizes('longPrimer', script);

export const getBrevier = script => getTypeSizes('brevier', script);

export const getMinion = script => getTypeSizes('minion', script);

export const getAtlas = script => getTypeSizes('atlas', script);

export const getElephant = script => getTypeSizes('elephant', script);

export const getImperial = script => getTypeSizes('imperial', script);

export const getRoyal = script => getTypeSizes('royal', script);

export const getFoolscap = script => getTypeSizes('foolscap', script);

export const getBodyCopy = script => getTypeSizes('bodyCopy', script);

/* ORIGINAL CONSTANTS */
export const GEL_ATLAS = `
  font-size: 4.875rem;
  line-height: 5.25rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    font-size: 6rem;
    line-height: 6.5rem;
  }

  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 8.75rem;
    line-height: 9.25rem;
  }
`;

export const GEL_ELEPHANT = `
  font-size: 3.75rem;
  line-height: 4rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    font-size: 4.875rem;
    line-height: 5.25rem;
  }

  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 7.25rem;
    line-height: 7.75rem;
  }
`;

export const GEL_IMPERIAL = `
  font-size: 3.125rem;
  line-height: 3.375rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    font-size: 4rem;
    line-height: 4.5rem;
  }

  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 6rem;
    line-height: 6.5rem;
  }
`;

export const GEL_ROYAL = `
  font-size: 2.5rem;
  line-height: 2.75rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    font-size: 3.25rem;
    line-height: 3.75rem;
  }

  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 4.75rem;
    line-height: 5.25rem;
  }
`;

export const GEL_FOOLSCAP = `
  font-size: 2rem;
  line-height: 2.25rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    font-size: 2.5rem;
    line-height: 2.75rem;
  }

  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 3.5rem;
    line-height: 3.75rem;
  }
`;

export const GEL_CANON = `
  font-size: 1.75rem;
  line-height: 2rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    font-size: 2rem;
    line-height: 2.25rem;
  }

  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 2.75rem;
    line-height: 3rem;
  }
`;

export const GEL_TRAFALGAR = `
  font-size: 1.25rem;
  line-height: 1.5rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    font-size: 1.5rem;
    line-height: 1.75rem;
  }

  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 2rem;
    line-height: 2.25rem;
  }
`;

export const GEL_PARAGON = `
  font-size: 1.25rem;
  line-height: 1.5rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    font-size: 1.375rem;
    line-height: 1.625rem;
  }

  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 1.75rem;
    line-height: 2rem;
  }
`;

export const GEL_DOUBLE_PICA = `
  font-size: 1.25rem;
  line-height: 1.5rem;

  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 1.5rem;
    line-height: 1.75rem;
  }
`;

export const GEL_GREAT_PRIMER = `
  font-size: 1.125rem;
  line-height: 1.375rem;

  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
`;

export const GEL_BODY_COPY = `
  font-size: 0.9375rem;
  line-height: 1.25rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_AND_LARGER} {
    font-size: 1rem;
    line-height: 1.375rem;
  }
`;

export const GEL_PICA = `
  font-size: 0.9375rem;
  line-height: 1.25rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_AND_LARGER} {
    font-size: 1rem;
    line-height: 1.25rem;
  }
`;

export const GEL_LONG_PRIMER = `
  font-size: 0.9375rem;
  line-height: 1.125rem;

  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 0.875rem;
  }
`;

export const GEL_BREVIER = `
  font-size: 0.875rem;
  line-height: 1rem;

  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    line-height: 1.125rem;
  }

  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 0.8125rem;
  }
`;

export const GEL_MINION = `
  font-size: 0.75rem;
  line-height: 1rem;
`;
