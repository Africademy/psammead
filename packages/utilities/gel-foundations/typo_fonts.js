module.exports = require('./dist/typo_fonts'); // eslint-disable-line import/no-unresolved
