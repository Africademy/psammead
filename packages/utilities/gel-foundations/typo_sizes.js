module.exports = require('./dist/typo_sizes'); // eslint-disable-line import/no-unresolved
