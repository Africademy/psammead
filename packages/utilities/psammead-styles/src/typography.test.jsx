import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { latin } from '@bbc/gel-foundations/scripts';
import { getCanon } from '@bbc/gel-foundations/typography';
import TypographyText from './typography';

describe('TypographyText', () => {
  shouldMatchSnapshot(
    'should render paragraph text with specified script and typography',
    <TypographyText script={latin} typographyFunc={getCanon}>
      Some text.
    </TypographyText>,
  );

  shouldMatchSnapshot(
    'should render paragraph text with no styling if script and typography function are not provided',
    <TypographyText>Some text.</TypographyText>,
  );
});
