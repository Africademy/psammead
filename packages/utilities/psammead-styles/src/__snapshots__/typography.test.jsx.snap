// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypographyText should render paragraph text with no styling if script and typography function are not provided 1`] = `
<p
  className=""
>
  Some text.
</p>
`;

exports[`TypographyText should render paragraph text with specified script and typography 1`] = `
.c0 {
  font-size: 1.75rem;
  line-height: 2rem;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c0 {
    font-size: 2rem;
    line-height: 2.25rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    font-size: 2.75rem;
    line-height: 3rem;
  }
}

<p
  className="c0"
>
  Some text.
</p>
`;
